patches/tModLoader/Terraria/Wiring.cs.patch

```diff
@@ -74,6 +74,24 @@
  	public static void SkipWire(Point16 point)
  	{
  		_wireSkip[point] = true;
+@@ -193,11 +_,17 @@
+ 		}
+ 	}
+ 
++	/// <summary>
++	/// Contains shared code for tile interactions. Usually called on the server and all clients. Typically called in conjunction with <c>NetMessage.SendData(MessageID.HitSwitch...)</c> to have the method run everywhere else.
++	/// </summary>
+ 	public static void HitSwitch(int i, int j)
+ 	{
+ 		if (!WorldGen.InWorld(i, j) || Main.tile[i, j] == null)
+ 			return;
+ 
++		if (!TileLoader.HitSwitch(i, j, Main.tile[i, j].TileType))
```

Comment:
The docs mention something about the return value of `HitSwitch` 

```
Return true to indicate that the tile had some sort of interaction occur. This return value is only used to force specific friendly NPC to keep walking, preventing them from resting on pressure plates, for example.
```

But I don't see that reflected in the code here

Reply:
That's the `ModTile.SwitchTiles` hook docs. It's in `NPC.UpdateNPC_Inner` where `Collision.SwitchTiles` is called.